File: go/raindrops/raindrops.go
package raindrops

import "fmt"

const testVersion = 2

// Convert turns an int (num) into a string of raindrop sounds if divisible by 3, 5, or 7, and returns the string version of itself if not
func Convert(num int) string {
	out := ""
	if num%3 == 0 {
		out += "Pling"
	}

	if num%5 == 0 {
		out += "Plang"
	}

	if num%7 == 0 {
		out += "Plong"
	}

	if out == "" {
		out = fmt.Sprintf("%v", num)
	}
	return out
}

File: ruby/raindrops/raindrops.rb
class Raindrops
  attr_reader :number
  def self.convert(arg1)
    Raindrops.new(number: arg1).response
  end

  def initialize(args)
    @number = args[:number]
  end

  def response
    response = ''
    response_table.each do |key, value|
      response << value if (number % key).zero?
    end
    response.empty? ? number.to_s : response
  end

  def response_table
    { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }
  end

end

class BookKeeping
  VERSION = 2
end
